reconcile/ocm_labels/integration.py

```diff
@@ -0,0 +1,392 @@
+        return organization_query(query_func).organizations or []
+
+    def get_environments(self, query_func: Callable) -> list[OCMEnvironment]:
+        return ocm_environment_query(query_func).environments
+
+    def init_ocm_apis(
+        self,
+        environments: Iterable[OCMEnvironment],
+        init_ocm_base_client: Callable[
+            [OCMAPIClientConfigurationProtocol, SecretReaderBase], OCMBaseClient
+        ] = init_ocm_base_client,
+    ) -> dict[str, OCMBaseClient]:
+        return {
+            env.name: init_ocm_base_client(env, self.secret_reader)
+            for env in environments
+        }
+
+    def get_early_exit_desired_state(self) -> Optional[dict[str, Any]]:
+        # gqlapi = gql.get_api()
+        # return {"clusters": [c.dict() for c in self.get_clusters(gqlapi.query)]}
+        return {}
+
+    def run(self, dry_run: bool) -> None:
+        gqlapi = gql.get_api()
+        clusters = self.get_clusters(gqlapi.query)
+        organizations = self.get_organizations(gqlapi.query)
+        environments = self.get_environments(gqlapi.query)
+
+        self.ocm_apis = self.init_ocm_apis(environments, init_ocm_base_client)
+
+        # organization labels
+        orgs_current_state, orgs_desired_state = self.fetch_organization_label_states(
+            organizations
+        )
+        self.reconcile(
+            dry_run=dry_run,
+            scope="organization",
+            current_state=orgs_current_state,
+            desired_state=orgs_desired_state,
+        )
+
+        # subscription labels
+        subs_current_state, subs_desired_state = self.fetch_subscription_label_states(
+            clusters
+        )
+        self.reconcile(
+            dry_run=dry_run,
+            scope="cluster",
+            current_state=subs_current_state,
+            desired_state=subs_desired_state,
+        )
+
+    def fetch_organization_label_states(
+        self, organizations: Iterable[OpenShiftClusterManagerV1]
+    ) -> tuple[
+        dict[LabelOwnerRef, dict[str, str]], dict[LabelOwnerRef, dict[str, str]]
+    ]:
+        """
+        Returns the current and desired state of the organizations labels for
+        the given organizations.
+
+        Please note that the current state might not contain all requested organizations,
+        e.g. if a organization can't be found in OCM.
+        """
+        label_sources: list[LabelSource] = [
+            init_aus_org_label_source(gql.get_api().query),
+        ]
+        current_state = self.fetch_organization_label_current_state(
+            organizations, self.params.managed_label_prefixes
+        )
+        desired_state = self.fetch_desired_state(label_sources)
+        return current_state, desired_state
+
+    def fetch_organization_label_current_state(
+        self,
+        organizations: Iterable[OpenShiftClusterManagerV1],
+        managed_label_prefixes: list[str],
+    ) -> dict[LabelOwnerRef, dict[str, str]]:
```

Comment:
An alias for `dict[LabelOwnerRef, dict[str, str]]` would be easier to read and can be reused

Reply:
done